File: go-solutions/1456_max_vowels_test.go
package leetcode

import "testing"

type maxVowels struct {
	s    string
	k    int
	want int
}

var maxVowelsTests = []maxVowels{
	{"abciiidef", 3, 3},
	{"aeiou", 2, 2},
	{"leetcode", 3, 2},
}

func TestMaxVowels(t *testing.T) {
	for _, test := range maxVowelsTests {
		if got := MaxVowels(test.s, test.k); got != test.want {
			t.Errorf("got %d want %d given: s = %q, k = %d", got, test.want, test.s, test.k)
		}
	}
}

File: ts-solutions/tests/1456_max_vowels.test.ts
import { maxVowels } from '../src/1456_max_vowels'

type MaxVowelsTest = {
	s: string
	k: number
	want: number
}

const tests: MaxVowelsTest[] = [
	{ s: `abciiidef`, k: 3, want: 3 },
	{ s: `aeiou`, k: 2, want: 2 },
	{ s: `leetcode`, k: 3, want: 2 },
]

describe(`maxVowels`, () => {
	test.each(tests)(`maxVowels($s, $k) => $want`, ({ s, k, want }) => {
		expect(maxVowels(s, k)).toEqual(want)
	})
})
